frontend/src/utils/date-utils.ts

```diff
@@ -100,22 +114,46 @@ export const dateStrToUpperCase = (dateStr: string): string => {
 };
 
 export const getDateFormat = (
-  date: number | string | undefined,
-  format: 'default' | 'snake',
+  date: number | string | undefined | Date,
+  format:
+    | 'default'
+    | 'snake'
+    | typeof DEFAULT_DATE_FORMAT_SNAKE_CASE
+    | typeof DEFAULT_DATE_FORMAT
+    | 'YYYY-MM-DD HH:mm'
+    | 'DD_MM_YYYY'
+    | 'YYYY-MM-DDTHH:mm:ss',
 ) => {
   if (date === undefined) {
     return undefined;
   }
 
+  const jsDate = new Date(date);
+  const year = jsDate.getUTCFullYear();
+  const month = String(jsDate.getUTCMonth() + 1).padStart(2, '0');
+  const day = String(jsDate.getUTCDate()).padStart(2, '0');
+  const hours = String(jsDate.getUTCHours()).padStart(2, '0');
+  const minutes = String(jsDate.getUTCMinutes()).padStart(2, '0');
+  const seconds = String(jsDate.getUTCSeconds()).padStart(2, '0');
+
   switch (format) {
     case 'default':
-      return moment(date).format(DEFAULT_DATE_FORMAT);
+    case DEFAULT_DATE_FORMAT:
+      return `${year}-${month}-${day}`;
     case 'snake':
-      return moment(date).format(DEFAULT_DATE_FORMAT_SNAKE_CASE);
+    case DEFAULT_DATE_FORMAT_SNAKE_CASE:
+      return `${year}_${month}_${day}`;
+    case 'YYYY-MM-DD HH:mm':
+      return `${year}-${month}_${day} ${hours}:${minutes}`;
+    case 'DD_MM_YYYY':
+      return `${day}_${month}_${year}`;
+    case 'YYYY-MM-DDTHH:mm:ss':
+      return `${year}-${month}-${day}T${hours}:${minutes}:${seconds}`;
```

Comment:
Did you look into `date-fns`? Seems like we're reinventing the wheel a bit here no?

Reply:
This is as vanilla javascript and as simple as it gets. If you want we can use `date-fns`